pkg/provider/provider_helpers.go

```diff
@@ -88,6 +44,45 @@ func toProtocol(s string) (protocol, error) {
 	}
 }
 
+type driverLogLevel string
+
+const (
+	logLevelOff driverLogLevel = "OFF"
+	// The levels below are lowercase on purpose to match the driver docs.
+	logLevelError driverLogLevel = "error"
+	logLevelWarn  driverLogLevel = "warn"
+	logLevelPrint driverLogLevel = "print"
+	logLevelTrace driverLogLevel = "trace"
+	logLevelDebug driverLogLevel = "debug"
+	logLevelInfo  driverLogLevel = "info"
+)
+
+var allDriverLogLevels = []driverLogLevel{
+	logLevelOff,
+	logLevelError,
+	logLevelWarn,
+	logLevelPrint,
+	logLevelTrace,
+	logLevelDebug,
+	logLevelInfo,
+}
+
+func toDriverLogLevel(s string) (driverLogLevel, error) {
+	s = strings.ToUpper(s)
+	switch s {
```

Comment:
This doesn't seem right. With this code every log level is invalid except for OFF which is upper case. Every other log level should be lowercase, but `return driverLogLevel(s), nil` will return all values with upper case.

Reply:
\+ you can add test for this

Reply:
Reworked after confirming with the driver team. Added tests.